codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolVisitor.kt

```diff
@@ -259,12 +259,7 @@ class SymbolVisitor(
 
     override fun setShape(shape: SetShape): Symbol {
         val inner = this.toSymbol(shape.member)
-        val builder = if (model.expectShape(shape.member.target).isStringShape) {
-            symbolBuilder(shape, RustType.HashSet(inner.rustType()))
-        } else {
-            // only strings get put into actual sets because floats are unhashable
-            symbolBuilder(shape, RustType.Vec(inner.rustType()))
-        }
```

Comment:
let's a comment that justifies why this is OK—when this code was written, sets could contain floats but that was changed:

> A set shape requires a single member named member, and the member MUST target either a string, blob, byte, short, integer, long, bigInteger, or bigDecimal shape. The targeted shape MUST NOT be marked with the [streaming trait](https://awslabs.github.io/smithy/1.0/spec/core/stream-traits.html#streaming-trait).